product-plane-services/devops-server/src/main/java/org/opendatamesh/platform/pp/devops/server/services/PolicyServiceProxy.java

```diff
@@ -100,24 +108,23 @@ public boolean isCallbackResultValid(TaskResultResource taskResult) {
         }
     }
 
-    public boolean isContextuallyCoherent(ActivityResource activity) {
+    public boolean isContextuallyCoherent(ActivityResource activityResource, DataProductVersionDPDS dataProductVersionDPDS) {
         if (!policyServiceActive) {
             logger.info("Policy Service is not active;");
             return true;
         }
         try {
             PolicyEvaluationRequestResource evaluationRequest = new PolicyEvaluationRequestResource();
-            evaluationRequest.setEvent(PolicyEvaluationRequestResource.EventType.TASK_EXECUTOR_FINAL_CALL);
-            evaluationRequest.setResourceType(PolicyEvaluationRequestResource.ResourceType.ACTIVITY);
-            evaluationRequest.setCurrentState(objectMapper.writeValueAsString(activity));
-
+            evaluationRequest.setEvent(PolicyEvaluationRequestResource.EventType.ACTIVITY_EXECUTION_RESULT);
+            evaluationRequest.setResourceType(PolicyEvaluationRequestResource.ResourceType.ACTIVITY_RESULT);
+            evaluationRequest.setCurrentState(JsonNodeMapper.toJsonNode(
+                    eventTypeMapper.toResource(activityResource, dataProductVersionDPDS)
+            ));
             ValidationResponseResource evaluationResult = policyClient.validateInputObject(evaluationRequest);
             if (Boolean.FALSE.equals(evaluationResult.getResult())) {
                 logger.warn("Policy evaluation failed during context coherence validation.");
             }
             return evaluationResult.getResult();
-        } catch (JsonProcessingException e) {
```

Comment:
Idem qui, magari rilancia una 500

Reply:
in tutti questi casi viene rilanciata l'eccezione generata da JsonNodeMapper, ma se preferisci possiamo ri-gestire anche quella specifica al posto che incapsularla dentro il catch normale (al momento verrebbe aggiunta come causa dell'eccezione catchata)